### ISSUE TITLE
Lost a padding in inceptionv4

### ISSUE THREAD

**Issue by YouhuiBai** [osmr/imgclsmob#21]
Hello, I tried to run inception-v4 on our local machine, found that the output size of average pool after the fully connected layer is not correct, I checked the source code, found that there lost a parameter `padding=1` in nn.AvgPool2D() function. Is it right?

**Comment by osmr** [comment]
I think you mean the Gluon version of the model.

If you want to run the model on images of arbitrary size, you must change that AvgPool2D to GlobalAvgPool2D or AdaptiveAvgPooling2D. It can be done on the fly. Or you can apply this model to 299x299 input:)

The original model uses global average pooling without paddings (see https://github.com/tensorflow/models/blob/master/research/slim/nets/inception_v4.py).

During the implementation of the model, I was guided by the Cadene's implementation (https://github.com/Cadene/pretrained-models.pytorch/blob/master/pretrainedmodels/models/inceptionv4.py) rather than the MXNet's one (https://github.com/apache/incubator-mxnet/blob/master/example/image-classification/symbols/inception-v4.py).

**Comment by YouhuiBai** [comment]
@osmr Thanks a lot for your recommendation, I really compared your inception-v4 implementation with Mxnet's one. I will try to replace AvgPool2D with GlobalAvgPool2D.

**Comment by osmr** [comment]
Fine. In your case, you can do the following:
```
from gluoncv2.model_provider import get_model as glcv2_get_model
import mxnet as mx
net = glcv2_get_model("inceptionv4", pretrained=True)
net.features._children.popitem()
net.features.add(mx.gluon.nn.GlobalAvgPool2D())
net.hybridize()
x = mx.nd.random.normal(shape=(1, 3, 224, 224), ctx=mx.cpu())
y = net(x)
```

**Comment by YouhuiBai** [comment]
@osmr All right, I will try later, thank you very much.
